pkg/provider/openshiftversionprovider.go

```diff
@@ -0,0 +1,85 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/blang/semver/v4"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kong/kubernetes-telemetry/pkg/types"
+)
+
+const (
+	// OpenShiftVersionKey is a report key used to report the OpenShift version, if any.
+	OpenShiftVersionKey = types.ProviderReportKey("openshift_version")
+	// OpenShiftVersionKind is the OpenShift cluster version kind.
+	OpenShiftVersionKind = Kind(OpenShiftVersionKey)
+
+	// OpenShiftVersionPodNamespace is a namespace expected to contain pods whose environment includes OpenShift version
+	// information.
+	OpenShiftVersionPodNamespace = "openshift-apiserver-operator"
+	// OpenShiftVersionPodApp is a value for the "app" label to select pods whose environment includes OpenShift version
+	// information.
+	OpenShiftVersionPodApp = "openshift-apiserver-operator"
+
+	// ImageVersionVariable is the environment variable whose value contains the operator image version. For OpenShift
+	// operators, this matches the OpenShift version.
+	ImageVersionVariable = "OPERATOR_IMAGE_VERSION"
+)
+
+// NewOpenShiftVersionProvider creates telemetry data provider that will query the
+// configured k8s cluster - using the provided client - to get cluster k8s version.
+func NewOpenShiftVersionProvider(name string, kc kubernetes.Interface) (Provider, error) {
+	return NewK8sClientGoBase(name, OpenShiftVersionKind, kc, openShiftVersionReport)
+}
+
+// openShiftVersionReport prepares a report that indicates the OpenShift version.
+func openShiftVersionReport(ctx context.Context, kc kubernetes.Interface) (types.ProviderReport, error) {
+	r := make(types.ProviderReport)
+
+	version, found := detectOpenShiftVersion(ctx, kc)
+	if !found {
+		return r, nil
+	}
+	return osVersionReport(version), nil
+}
+
+// detectOpenShiftVersion checks for the presence of a known OpenShift Pod and obtains the OpenShift version from it.
+// It returns the version (which may be empty) and a boolean indicating if the Pod was found at all. If it didn't find
+// the Pod, it's probably not OpenShift.
+func detectOpenShiftVersion(ctx context.Context, kc kubernetes.Interface) (semver.Version, bool) {
+	var pods []corev1.Pod
+	cont := ""
```

Comment:
Nit: we could either use `:=` or `var` (and long notation) but let's be consistent.

```suggestion
	var (
		pods []corev1.Pod
		cont string
	)
```